tt_metal/detail/tt_metal.hpp

```diff
@@ -368,7 +369,10 @@ namespace tt::tt_metal{
                 pcie_chan_end_addr += tt::Cluster::instance().get_host_channel_size(device->id(), pcie_chan);
             }
 
-            CoreCoord enqueue_program_dispatch_core = *device->consumer_cores().begin();
+            uint16_t channel = tt::Cluster::instance().get_assigned_channel_for_device(device->id());
+            const uint8_t cq_id = 0; // Currently, only the first command queue is responsible for enqueuing programs
+            tt_cxy_pair enqueue_program_dispatch_core = dispatch_core_manager::get(device->num_hw_cqs()).command_dispatcher_core(device->id(), channel, cq_id);
+            CoreCoord physical_enqueue_program_dispatch_core = get_physical_dispatch_coord(enqueue_program_dispatch_core);
```

Comment:
can you this logic to fetch `physical_enqueue_program_dispatch_core` into `jit_build_genfiles_noc_addr_ranges_header` (or lower)? This isn't used anywhere else in this function